### PR TITLE
JSONI8 (Aseprite's json format) Support added (NO MORE JSONTOXML CONVERTERS)

### PR DESCRIPTION
**Pull request by moondroidcoder**
I've added support for aseprite's json (called JSONI8), its very useful for pixel artists who use aseprite, since you wouldnt need to convert the I8 sprite sheet into an xml or someshit. Ive added an example_bf for people to base their I8 character off aswell.

I will make a tutorial on how to add I8 sprite sheets soon. so far this is only for characters, I will add support for luasprites later on

### CODE REVIEW

**Comment by zacksgamerz** [comment]
Oh and also credits:
me ( https://twitter.com/ZackDroidCoder ) and my pal luckydog ( https://www.youtube.com/channel/UCeHXKGpDKo2eqYKVkqCUdaA ) made the code for this
rusron made the example-bf sprite https://twitter.com/Rusron1

**Comment by zacksgamerz** [comment]
To make an lua sprite using the new json format, just basically make it like any lua sprite `makeAnimatedLuaSprite(tag, image, x, y, 'jsoni8')` make sure the last argument is `i8` or `json` or `jsoni8` and then you can add it like any other sprite

**Comment by zacksgamerz** [comment]
image
here is a more thorough mini-explanation thatd help aswell

**Comment by zacksgamerz** [comment]
you wont have to use some converter anymore !

**Comment by windows10win7posreadywoeskio** [comment]
YEEEEEEEEEEEEEEEEEES

**Comment by TylerTehIdiot** [comment]
epic

**Comment by KadeDev** [comment]
would this not be better if you just made a json8 to xml converter in psych?

**Comment by zacksgamerz** [comment]
this goes without saying but duration and slices are advanced, so Ive decided to leave them for later since I myself dont know how they work yet

**Comment by KadeDev** [comment]
> > would this not be better if you just made a json8 to xml converter in psych?
> 
> imo the only difference takes effect in the source, which I see isnt that necessary when the user who's using the modding system just needs to export the sprite sheet, and put it in the game like any other sprite sheet, the only difference is that its an aseprite atlas and not sparrow atlas which isnt that big of a deal when it comes to the user adding the sprite in.
> 
> Same with source coding, all you have to do is `sprite.frames = Paths.fromI8("spriteName", "shared");`
> 
> plus itd save trouble for the user in the future, since there are some arguments in the json that arent added yet, like fps rate, duration, and slices. which are kinda for the lads who have experience in aseprite's sheet formatting. Lmk if I missed anything! :)

yet again, I just feel like it'd be better adding support to convert it to the main format the game already uses rather than support a new one.

**Review by TheKitBoi — APPROVED** [review]

**Comment by Stilic** [comment]
> would this not be better if you just made a json8 to xml converter in psych?

sorry kade im lazy this is a shitty idea

**Comment by DragShot** [comment]
> would this not be better if you just made a json8 to xml converter in psych?

I am certain such a tool already exists: https://github.com/Lethrial/Aseprite-.json-to-FNF-.xml

**Comment by zacksgamerz** [comment]
A little side note: Since this isnt a converter, this gives more accessibility to longtime aseprite users in the future like I said. The main purpose of this is for pixel artists to not go through the trouble to convert their json (removing its extra features aswell) into xml by downloading tools, comparing or suggesting it to be a converter just defeats the main goal of why me and lucky coded this. Any feedback is appreciated though! :D

**Comment by PrismLight** [comment]
Cool, but can't you just make a haxelib for Aseprite support?

**Comment by zacksgamerz** [comment]
I will once I finish the advanced features! But you see, this gives support to psych engine's modding system and lua system too, so its basically 2 in 1

**Comment by PrismLight** [comment]
> > Cool, but can't you just make a haxelib for Aseprite support?
> 
> I will once I finish the advanced features! But you see, this gives support to psych engine's modding system and lua system too, so its basically 2 in 1

Makes sense.

**Comment by zacksgamerz** [comment]
small merge conflict fix

**Comment by zacksgamerz** [comment]
another merge conflict fix that i missed

File: source/FunkinLua.hx [hunk 1]
```diff
@@ -1377,6 +1377,16 @@ class FunkinLua {
 			leSprite.antialiasing = ClientPrefs.globalAntialiasing;
 			PlayState.instance.modchartSprites.set(tag, leSprite);
 		});
+		
+		// Lua_helper.add_callback(lua, "makeJsonI8LuaSprite", function(tag:String, image:String, x:Float, y:Float, ?spriteType:String = "sparrow") {
```
Comment (ImaginationSuperHero528) [inline source/FunkinLua.hx:1381]: I recommend shorting it to `makeI8LuaSprite` or `makeI8Sprite`- idrk know at this point.

**Review by rodney528 — APPROVED** [review]

**Review by Joalor64 — APPROVED** [review]

File: source/FunkinLua.hx (hunk 1 above)
Comment (zacksgamerz) [reply source/FunkinLua.hx:1381]: > I recommend shorting it to `makeI8LuaSprite` or `makeI8Sprite`- idrk know at this point.

thats unused code lmao, it isnt even applied its commented

Comment (ImaginationSuperHero528) [reply source/FunkinLua.hx:1381]: > > I recommend shorting it to `makeI8LuaSprite` or `makeI8Sprite`- idrk know at this point.
> 
> thats unused code lmao, it isnt even applied its commented

I didn't even notice XD

**Review by bananaTiko — APPROVED** [review]
OH MY GOD YESSSS **I COULD KISS YOU ON THE MOUTH**

**Comment by ShadowMario** [comment]
No longer needed since the last experimental commit

**Pull request closed without merging**